bootloader/src/pyi_utils.c

```diff
@@ -266,30 +310,36 @@ pyi_get_temp_path(char *buffer, char *runtime_tmpdir)
     wchar_t prefix[16];
     wchar_t wchar_buffer[PATH_MAX];
     char *original_tmpdir;
-    char runtime_tmpdir_abspath[PATH_MAX + 1];
+    wchar_t *wruntime_tmpdir_abspath;
+    DWORD rc;
 
     if (runtime_tmpdir != NULL) {
-      /*
-       * Get original TMP environment variable so it can be restored
-       * after this is done.
-       */
-      original_tmpdir = pyi_getenv("TMP");
-      /*
-       * Set TMP to runtime_tmpdir for _wtempnam() later
-       */
-      pyi_path_fullpath(runtime_tmpdir_abspath, PATH_MAX, runtime_tmpdir);
-      pyi_setenv("TMP", runtime_tmpdir_abspath);
+        // Get original TMP environment variable so it can be restored after this is done.
+        original_tmpdir = pyi_getenv("TMP");
```

Comment:
As you are here, I am not very comfortable with the envar name. How about a more PyInstaller specific one, like `PYI_TMP` or `PYI_RUNTIME_TMPPATH`?
```suggestion
        original_tmpdir = pyi_getenv("TMP");
```
WDYT @rokm?

Reply:
I believe the function used below, `GetTempPathW` and `wtempnam`, relies on having either `%TMP%` or `%TEMP%` set in Windows.  [Documentation for GetTempPathW here](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-gettemppathw).

However, just from a quick glance, when using the `--runtime-tmp-dir` option, I'm not sure we even really need to call `GetTempPathW` since we already have the full path to where we want to unpack.  In this case, we could avoid using environment variables.  According to the documentation for [_wtempnam](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/tempnam-wtempnam-tmpnam-wtmpnam?view=vs-2019), it seems we can pass in the directory path we want to use instead of using the value of `%TMP%`.

So in conclusion, I think we could get rid of manipulating environment variables here like so:
```
// Psuedo-code
if (runtime_tmpdir != NULL) {
  // Don't use %TMP%
  tmpdir = pyi_build_temp_folder(runtime_tmpdir);
} else {
  // Use %TMP%
  tmpdir = GetTempPathW()
}
...
wchar_ret = _wtempnam(buffer, tmpdir)
```
Thoughts?  I would be happy to implement these changes if the logic seems valid.

Reply:
It seems a good point, let's see if it works smoothly :)

Reply:
@BoboTiG it seems to me that the idea in the existing code) is indeed to temporarily override `TMP` variable for `GetTempPathW()` and `_wtempnam()˙ calls and then restore it afterwards.

Regarding the proposed amendment; does skipping `GetTempPathW()` if we have `runtime_tmpdir` gain us anything? Is it more reliable (e.g., could `GetTempPathW()` return something else than the `TMP` that we set)? 

If not, I don't see any benefit in additional distinction, because we need to set `TMP` for subsequent `_wtempnam()` calls anyway (otherwise, if I understand documentation correctly, if `TMP` is set in the environment, the generated names will be unique to `TMP` instead of the `runtime_tmpdir`).

Reply:
OK you are right. Bad idea at the beginning, sorry both of you :)